### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by oleksandr-lysanets**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,10 +1,53 @@
+import json
+
 import init_django_orm  # noqa: F401
 
 from db.models import Race, Skill, Player, Guild
 
 
+with open("players.json") as players:
+    players_data = json.load(players)
```
Comment (Bohdan-Salamakha) [inline main.py:9]: place it into a separated function

File: main.py [hunk 2]
```diff
@@ -1,10 +1,53 @@
...
+            else:
+                guild = Guild.objects.get(name=data["guild"]["name"])
+
+        Player.objects.create(
+            nickname=name,
+            email=data["email"],
+            bio=data["bio"],
+            race=race,
+            guild=guild
+        )
```
Comment (Bohdan-Salamakha) [inline main.py:50]: separate your function into a few functions with own logic

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
v1

File: main.py [hunk 3]
```diff
@@ -1,10 +1,69 @@
...
+
+def create_skill(data: dict) -> None:
+    for skill in data["race"]["skills"]:
+        if not Skill.objects.filter(name=skill["name"]).exists():
+            skill = Skill.objects.create(
+                name=skill["name"],
+                bonus=skill["bonus"]
+            )
+        else:
+            skill = Skill.objects.get(name=skill["name"])
```
Comment (olena-hul) [inline main.py:32]: Use `.get_or_create` method for this

Comment (olena-hul) [reply main.py:32]: https://docs.djangoproject.com/en/4.1/ref/models/querysets/#get-or-create

Comment (olena-hul) [reply main.py:32]: Fix in all places where needed

**Review by olena-hul — CHANGES_REQUESTED** [review]

File: main.py [hunk 4]
```diff
@@ -1,10 +1,60 @@
...
+    return race
+
+
+def create_skill(data: dict) -> None:
+    for skill in data["race"]["skills"]:
+        skill, _ = Skill.objects.get_or_create(
+            name=skill["name"],
+            bonus=skill["bonus"]
+        )
+        skill.race = create_race(data)
```
Comment (olena-hul) [inline main.py:27]: Create race once at the begginning of the function
And then just assign the value in a loop

This way you'll reduce some queries to the database

**Review by olena-hul — CHANGES_REQUESTED** [review]

**Review by olena-hul — APPROVED** [review]

**Comment by olena-hul** [comment]
Good job!

**Pull request closed without merging**